app/src/main/java/org/onionshare/android/ui/settings/SettingsTorUi.kt

```diff
@@ -0,0 +1,287 @@
+    Text(
+        modifier = Modifier
+            .padding(horizontal = 16.dp),
+        text = stringResource(R.string.settings_tor_bridges_intro),
+        style = MaterialTheme.typography.body2,
+    )
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(horizontal = 16.dp),
+        verticalAlignment = CenterVertically,
+    ) {
+        Text(
+            modifier = Modifier
+                .clickable {
+                    val linkIntent = Intent().apply {
+                        action = ACTION_VIEW
+                        data = Uri.parse(BRIDGE_DB)
+                    }
+                    context.startActivity(Intent.createChooser(linkIntent, null))
+                }
+                .weight(1f)
+                .padding(top = 8.dp, bottom = 8.dp, end = 8.dp),
+            text = BRIDGE_DB,
+            fontFamily = FontFamily.Monospace,
+            color = MaterialTheme.colors.OnionBlue,
+        )
+        ShareButton(BRIDGE_DB)
+    }
+    var textState by remember { mutableStateOf(TextFieldValue("")) }
+    OutlinedTextField(
+        modifier = Modifier
+            .padding(horizontal = 16.dp)
+            .padding(top = 8.dp)
+            .fillMaxWidth()
+            .heightIn(min = 128.dp),
+        value = textState,
+        onValueChange = { textState = it },
+        singleLine = false,
+    )
+    Row(
+        modifier = Modifier.fillMaxWidth(),
+    ) {
+        val clipboardManager = LocalClipboardManager.current
+        IconButton(
+            modifier = Modifier.padding(horizontal = 8.dp),
+            onClick = {
+                clipboardManager.getText()?.let {
+                    textState = TextFieldValue(it, selection = TextRange(it.length))
+                }
+            },
+        ) {
+            Icon(
+                imageVector = Icons.Filled.ContentPaste,
+                contentDescription = stringResource(R.string.paste),
+                tint = MaterialTheme.colors.onSurface.copy(alpha = 0.65f),
+            )
+        }
+        Spacer(modifier = Modifier.weight(1f))
+        TextButton(
+            modifier = Modifier.padding(horizontal = 16.dp),
+            onClick = { textState = TextFieldValue("") },
+            enabled = textState.text.isNotBlank(),
+        ) {
+            Text(stringResource(R.string.cancel))
+        }
+        TextButton(
+            modifier = Modifier.padding(horizontal = 16.dp),
+            onClick = {
+                onBridgesAdded(textState.text)
+                textState = TextFieldValue("")
+            },
+            enabled = textState.text.isNotBlank(),
+        ) {
+            Text(stringResource(R.string.add))
+        }
+    }
+}
+
+@Composable
+fun RadioPreference(
+    title: String,
+    summary: String,
+    selected: Boolean,
+    onSelected: () -> Unit,
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .selectable(
+                enabled = !selected,
```

Comment:
Is this disabling the radio button when it's not selected?

Reply:
Unfortunately not the radio button, but the text label of the button (making it non-selectable). I tried also disabling the radio button, but that shows visually which is not good.

Reply:
Sorry, I don't understand what it means for the text label to be disabled (therefore non-selectable?) when it's selected. How can it be non-selectable when it's selected, and why does this need to change based on whether it's selected?

Reply:
I found it odd that I could re-select the already selected option, causing a re-composition and ui update with it. You can still do it when select the radio circle directly.

Reply:
Interesting, thanks for explaining!